doc: Entity management API documentation (#3659)
thin-edge/thin-edge.io: docs/src/operate/registration/http_api.md

curl http://localhost:8000/te/v1/entities/device/child01
```

> ## Query entities

Author: The diff of this file is a bit weird. All I did was to move this `Query entities` section to the end of this file so that the CRUD operations are placed together. This `Query entities` section was quite big and in the middle of them earlier, placed along with `Fetch entities`(as querying is also another kind of READ operation).

An existing entity can be updated using the PATCH API.

> But the updates are limited to the `@parent` and `@health` endpoints only.
> Other properties like `@type` and `@id` can not be updated after the registration.

Reviewer: ```suggestion
Updates are limited to the `@parent` and `@health` endpoints only, so other properties like `@type` and `@id` cannot be updated after the registration.
```

The `@parent` must be another device, as services cannot have children.
Similarly, the `@health` endpoint must be a service, as only services publish their health statuses.

> The entities specified in either fields must be registered up-front, before they can be used.

Reviewer: It would be more helpful to give common uses of the api rather then going into the "limitations" of the api. e.g.

```
The update api is typically used to:
* change the parent of a child device
* change the health topic of a service
```

Author: Resolved by 4ff49e7

**Endpoint**

```
PATCH /v1/entities/{topic-id}
```

**Payload**

```json
{
    "@parent": "{new-parent-topic-id}",
    "@health": "{some-service-topic-id}"
}
```

**Example**

> 

Reviewer: A brief sentence explaining what the example is doing would be helpful for users to pick up on user-related purpose of the command quicker.

| `parent`  | Direct child entities of the given `parent` entity (excluding it)             | `device/main//`                     |
| `type`    | Entities of the given entity `type`                          | `main`, `child-device` or `service` |

The following restrictions apply:
* Multiple values can not be specified for the same parameter.
* The same parameter can not be repeated multiple times.
* The `root` and `parent` parameters can not be used together.

**Responses**

* 200: OK
  ```json
  [
      {
          "@topic-id": "device/main//",
          "@type": "device"
      },
      {
          "@topic-id": "device/main/service/service0",
          "@type": "service",
          "@parent": "device/main//"
      },
      {
          "@topic-id": "device/child0//",
          "@type": "child-device",
          "@parent": "device/main//"
      }
      ...
  ]
  ```
* 404: Not Found
  ```json
  {
      "error": "Entity with topic id: device/unknown// not found"
  }
  ```

**Examples**

To demonstrate different query examples, the following entity tree is assumed as the base:

> ```

Author: Resolved by [30a930c](https://github.com/thin-edge/thin-edge.io/pull/3659/commits/30a930ccd766937efb42c4102fbd619c376e1276)
